client/product_details/ExpandedView.jsx

```diff
@@ -1,24 +1,94 @@
 import React from 'react';
 
-const ExpandedView = (props) => {
+class ExpandedView extends React.Component {
+  constructor(props) {
+    super(props);
 
-  if (props.selectedStyle) {
-    var photoGallery = props.selectedStyle.photos;
-    return (
-      <div id="expandedView">
-        <a target="_blank">
-          <img 
-            src={photoGallery[props.currPhotoIndex].url || 'https://www.thermaxglobal.com/wp-content/uploads/2020/05/image-not-found.jpg'} 
-            alt={props.selectedStyle.name} 
-            className="defaultView" 
-            id={props.selectedStyle.style_id}>
-          </img>
-          <a onClick={props.toggleView} id="fullscreenIcon"><i class="fas fa-expand"></i></a>
-        </a>
-      </div>
-    );
-  } 
-  return null;
-};
+    this.state = {
+      zoomView: false
+    };
+    
+    this.toggleZoomView = this.toggleZoomView.bind(this);
+    this.handleMouseMove = this.handleMouseMove.bind(this);
+  }
+  
+  componentDidMount() {
+    var element = document.getElementById('expandedView');
+    console.log(element);
+    element.addEventListener("mousemove", this.handleMouseMove);
+  }
+
+  componentWillUnmount() {
+    var element = document.getElementById('expandedView');
+    console.log(element);
+    element.removeEventListener("mousemove", this.handleMouseMove);
+  }
+
+  toggleZoomView() {
+    this.setState({
+      zoomView: !this.state.zoomView
+    });
+  }
+  
+  handleMouseMove(e) {
+    console.log('in mouse move');
+    var element = document.getElementById('expandedView');
+    console.log('height: ', element.clientHeight);
+    console.log('width: ', element.clientWidth);
+    console.log('e background X: ', element.style.backgroundPositionX);
+    console.log('e background Y: ', element.style.backgroundPositionY);
+    console.log('e offset X: ', -e.offsetX);
+    console.log('e offset Y: ', -e.offsetY);
+    console.log('x%: ', (e.offsetX / element.clientWidth) * 100);
+    console.log('y%: ', (e.offsetY / element.clientHeight) * 100);
+    console.log('background size: ', element.style.backgroundSize);
+    var xPercent = (e.offsetX / element.clientWidth) * 100;
+    var yPercent = (e.offsetY / element.clientHeight) * 100;
+    element.style.backgroundPositionX = xPercent + "%";
+    element.style.backgroundPositionY = yPercent + "%";
+  }
```

Comment:
Aside from removing some of the console.log that you might no longer need, I just want to say that this looks awesome! I didn't know about all these properties before (e.g., offset and clientWidth/Height) Very neat how they're used together for this feature. I tried to do a little bit of reading on dimensions of elements to understand why you chose to use certain properties over the others, but I'm not going to lie that these little nuances still escaped me. Would be great if you could share some of the resources you used to understand this concept!